ONSdigital/sds
# Sdss 112 update firestore init param

### Motivation and Context
This change is made so that a mock google credential is not longer required to run the app in local environment

### What has changed
- Project ID is added as an environment variable for both local and cloud environment. Default to mock-project-id for the local environment while getting the project id from GCP for the cloud environment
- DATASET_BUCKET_NAME is added to the new_dataset cloud function as an environment variable
- Mock google credential is removed from environment variable of the local docker environment
- Firebase_Admin is no longer used to connect firebase client
- Classes BucketLoader and FirebaseLoader are added to establish the storage client and firebase client respectively, and both clients are connected using the Project ID environment variable.
- BucketLoader encapsulate the function to get Schema Bucket and Dataset Bucket
- FirebaseLoader encapsulate the function to get Schema Collection and Dataset Collection
- Both loaders will return None in an UnitTest environment
- Firebase Repository, Bucket Repository, and Integration Tests are modified to use the new loader classes
- After this change, functions to delete file from bucket and to delete previous version datasets from FireStore must be mocked in UnitTest whenever invoked or error will be prompted due to the return of None type from loaders
- [Fixes] Changed integration-test-cloudbuild to use variables supplied in cloudbuild.yaml instead of the global variables. This is to allow the cloud integration test to capture the correct Project ID, buckets name, API_URL from each sandbox, thus can pass the PR triggered cloud integration test on any sandbox

### How to test?
- Ensure app can function normally in the docker and sandbox environment
- Ensure unit test can function normally
- Ensure integration test can function normally in the docker and sandbox environment
- Ensure the new_dataset cloud function can run properly in sandbox
- Ensure pr-lint-checks are passed

### Links

### Screenshots (if appropriate):
